tools/image-audit/identity/imageidentity/urlandpageidentity.js

```diff
@@ -0,0 +1,317 @@
+      }
+    } catch (error) {
+      additionalTokensToSum.length = 0;
+      additionalTokensToSum.push('er');
+      additionalTokensToSum.push(site); // Start with the URL or other primary identifier
+    }
+
+    const { identityId } = await UrlIdentity.getUrlIdentityID(
+      clusterManager,
+      href,
+      originatingClusterId,
+      UrlAndPageIdentity.type,
+      additionalTokensToSum,
+    );
+    return identityId;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  async #obtainRum(values, identityState) {
+    try {
+      // TODO: Rum data should really be on a page cluster, not an image cluster.
+      // const siteURL = new URL(values.site);
+      const dataChunks = new DataChunks();
+
+      if (!identityState.rumLoadedData) {
+        if (identityState.rumLoadedError) {
+          return;
+        }
+        // TODO: Selectable range
+        const loader = new DataLoader();
+        loader.apiEndpoint = API_ENDPOINT;
+        loader.domain = values.replacementDomain;
+        loader.domainKey = values.domainKey;
+        if (identityState.rumLoadingPromise) {
+          try {
+            await identityState.rumLoadingPromise;
+          } catch (error) {
+            this.#rumData = {
+              pageViews: 0,
+              conversions: 0,
+              visits: 0,
+              bounces: 0,
+            };
+            return;
+          }
+        } else {
+          try {
+            identityState.rumLoadingPromise = loader.fetchPrevious12Months(null);
+            identityState.rumLoadedData = await identityState.rumLoadingPromise;
+            identityState.rumLoadingPromise = null;
+          } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error('Error loading rum data:', error);
+            // fill it with a stub so we dont slam RUM with requests.
+            identityState.rumLoadedError = true;
+            this.#rumData = {
+              pageViews: 0,
+              conversions: 0,
+              visits: 0,
+              bounces: 0,
+            };
+            return;
+          }
+        }
+      }
+      const { rumLoadedData } = identityState;
+
+      const conversionSpec = { checkpoint: ['click'] };
+
+      // set up metrics for dataChunks
+      dataChunks.addSeries('pageViews', (bundle) => bundle.weight);
+      dataChunks.addSeries('visits', (bundle) => (bundle.visit ? bundle.weight : 0));
+      // a bounce is a visit without a click
+      dataChunks.addSeries('bounces', (bundle) => (bundle.visit && !bundle.events.find(({ checkpoint }) => checkpoint === 'click')
+        ? bundle.weight
+        : 0));
+      dataChunks.addSeries('conversions', (bundle) => (dataChunks.hasConversion(bundle, conversionSpec)
+        ? bundle.weight
+        : 0));
+      dataChunks.addFacet('checkpoint', (bundle) => Array.from(bundle.events.reduce((acc, evt) => {
+        acc.add(evt.checkpoint);
+        return acc;
+      }, new Set())), 'every');
```

Comment:
Check https://github.com/adobe/helix-website/blob/main/tools/rum/slicer.js#L53-L63, the rum-distiller gives you the series (standardize definition).